align/pnr/checkers.py

```diff
@@ -297,7 +297,18 @@ def addt( obj, con, tag=None):
     for inst, parameters in subinsts.items():
         cnv.subinsts[inst].parameters.update(parameters)
 
-    nets_allowed_to_be_open = [] if toplevel else global_power_names
+    nets_dnr = []
```

Comment:
This is okay, but haven't we already read it the constraint file. Could we just pass it to this routine?

Reply:
I can pass 'pnr_const_ds' in 'generate_pnr' to '_generate_json', then to 'gen_viewer_json'.